src/apps/reducers.js

```diff
@@ -0,0 +1,59 @@
+import { UPDATE_LOCATION } from 'react-router-redux';
+import { combineReducers } from 'redux';
+import { NEW_BUNDLE, NEW_ERROR, FETCH_BUNDLE_ERROR } from './actions';
+
+function runMode(state = { path: null, rendezvous: null }, action) {
+    if (action.type === UPDATE_LOCATION) {
+        let { query } = action.payload;
+
+        if (query.path) {
```

Comment:
Very clean, but I was coding against the (unlikely) scenario that someone sets both path and rendezvous in the query string. Not quite sure how to prevent people from putting what ever the heck they want in the query.